components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/DashboardRestApi.java

```diff
@@ -141,16 +155,19 @@ public Response get(@PathParam("id") String id) {
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/")
-    public Response create(DashboardMetadata dashboardMetadata) {
+    public Response create(@Context Request request, DashboardMetadata dashboardMetadata) {
         try {
             if (!dashboardDataProvider.get(dashboardMetadata.getUrl()).isPresent()) {
-                dashboardDataProvider.add(dashboardMetadata);
+                dashboardDataProvider.add(request.getProperty(InterceptorConstants.PROPERTY_USERNAME).toString(),
```

Comment:
since `request.getProperty(InterceptorConstants.PROPERTY_USERNAME).toString()` is repeated many times, lets move it to a separate method. e.g. `private static String getUsername(request)`